### PR TITLE
4-WebApp-your-API/4-2-B2C does not mention the client secret #414

### PR DESCRIPTION
**Pull request by Shama-K**
## Purpose

Update Readme to Resolve Following GitHub Issue:
https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/issues/414

## Does this introduce a breaking change?

```
[ ] Yes
[ x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ x] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: 4-WebApp-your-API/4-2-B2C/README.md [hunk 1]
```diff
@@ -16,17 +17,12 @@ endpoint: Microsoft identity platform
 >
 > This article (README.md) contains the full instructions on how to configure the sample. If you have gone through the [first chapter](../1-WebApp-OIDC/1-1-MyOrg) and already configured the client web application to sigh-in users, read through the [README-incremental-instructions.md](README-incremental-instructions.md) instead.
```
Comment (kalyankrishna1) [inline 4-WebApp-your-API/4-2-B2C/README.md:18]: >sigh [](start = 216, length = 4)

sign

Comment (kalyankrishna1) [inline 4-WebApp-your-API/4-2-B2C/README.md:18]: Remove these lines

File: 4-WebApp-your-API/4-2-B2C/README.md [hunk 2]
```diff
@@ -16,17 +17,12 @@ endpoint: Microsoft identity platform
...
 
-## About this sample
+## Overview
 
-This sample is essentially a guide for developers who want to secure their Web APIs using the Microsoft identity platform (formerly Azure Active Directory for developers) using Azure AD B2C. This sample lays down the all the steps developers need to take to secure their web api with the Microsoft identity platform. Additionally it also explains the steps and processes for a client to obtain the necessary permissions and tokens to make successful calls to this secured web api.  
+This sample demonstrates a ASP.NET Core Web App application calling a ASP.NET Core Web API that is secured using Azure AD B2C.
```
Comment (kalyankrishna1) [inline 4-WebApp-your-API/4-2-B2C/README.md:22]: >a [](start = 25, length = 1)

an

File: 4-WebApp-your-API/4-2-B2C/README.md [hunk 3]
```diff
@@ -16,17 +17,12 @@ endpoint: Microsoft identity platform
...
-
-### Overview
-
-This sample presents a Web API running on ASP.NET Core, protected by [Azure AD OAuth Bearer](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-v2-protocols) Authentication. The client application uses [MSAL.NET](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet) library to obtain a JWT access token through using the [OAuth 2.0](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow) protocol flow.
+1. The client ASP.NET Core Web App application uses the Microsoft Authentication Library (MSAL) to obtain a JWT access token from **Azure AD B2C**:
```
Comment (kalyankrishna1) [inline 4-WebApp-your-API/4-2-B2C/README.md:24]: >Microsoft Authentication Library (MSAL) [](start = 56, length = 39)

link

File: 4-WebApp-your-API/4-2-B2C/README.md [hunk 4]
```diff
@@ -16,17 +17,12 @@ endpoint: Microsoft identity platform
...
+1. The access token is used as a bearer token to authenticate the user when calling the ASP.NET Core Web API.
```
Comment (kalyankrishna1) [inline 4-WebApp-your-API/4-2-B2C/README.md:25]: >access token [](start = 7, length = 12)

link

File: 4-WebApp-your-API/4-2-B2C/README.md (hunk 3 above)
Comment (kalyankrishna1) [inline 4-WebApp-your-API/4-2-B2C/README.md:24]: >to [](start = 96, length = 2)

to sign-in a user and

**Comment by kalyankrishna1** [comment]
> 1. Signs-in the user with local or social identities. When the user signs-in for the first time , a consent screen is presented. This consent screen lets the user consent for the application to access the web API( TodoListService).

B2C presents a consent screen? really?

---
Refers to: 4-WebApp-your-API/4-2-B2C/README.md:29 in ba50129. [](commit_id = ba50129f4afd0de7ee83e152e019fdc117f5fc8e, deletion_comment = False)

**Comment by kalyankrishna1** [comment]
> 1. Calls the Web API using the access token as a bearer token in the authentication header of the Http request. The Web API validates the caller using the ASP.NET JWT Bearer Authentication middleware.

authorization

---
Refers to: 4-WebApp-your-API/4-2-B2C/README.md:31 in ba50129. [](commit_id = ba50129f4afd0de7ee83e152e019fdc117f5fc8e, deletion_comment = False)

**Comment by kalyankrishna1** [comment]
> 1. Calls the Web API using the access token as a bearer token in the authentication header of the Http request. The Web API validates the caller using the ASP.NET JWT Bearer Authentication middleware.

authorization

---
Refers to: 4-WebApp-your-API/4-2-B2C/README.md:31 in ba50129. [](commit_id = ba50129f4afd0de7ee83e152e019fdc117f5fc8e, deletion_comment = False)

File: 4-WebApp-your-API/4-2-B2C/README.md [hunk 5]
```diff
@@ -36,26 +32,19 @@ The client web application essentially takes the following steps to sign-in the
...
+This sample has a web API and a client web app, both built using the asp.net core platform. The client app signs in users using the [OpenID Connect protocol](https://docs.microsoft.com/azure/active-directory/develop/v2-protocols-oidc) flow and in this process obtains (and caches) an [access token](https://docs.microsoft.com/azure/active-directory/develop/access-tokens) for the web API. The client app has a ToDo list that the web app users can work with. This ToDo list is maintained in an in-memory list on the Web API. The client app calls the web API for all operations on the ToDo list.
 
-## How to run this sample
+## Prerequisites
 
-### Pre-requisites
+- [Visual Studio](https://visualstudio.microsoft.com/downloads/)
+- An **Azure AD B2C** tenant. For more information see: [How to get an Azure AD B2C tenant](https://docs.microsoft.com/azure/active-directory-b2c/tutorial-create-tenant)
+- A user account in your **Azure AD B2C**. This sample will not work with a **personal Microsoft account**. Therefore, if you signed in to the [Azure portal](https://portal.azure.com) with a personal account and have never created a user account in your directory before, you need to do that now.
```
Comment (kalyankrishna1) [inline 4-WebApp-your-API/4-2-B2C/README.md:43]: >- A user account in your **Azure AD B2C**. This sample will not work with a **personal Microsoft account**. Therefore, if you signed in to the [Azure portal](https://portal.azure.com) with a personal account and have never created a user account in your directory before [](start = 0, length = 270)

remove this line

File: 4-WebApp-your-API/4-2-B2C/README.md (hunk 2 above)
Comment (kalyankrishna1) [inline 4-WebApp-your-API/4-2-B2C/README.md:22]: >a [](start = 68, length = 1)

an

**Comment by kalyankrishna1** [comment]
> This behavior is expected as you are not authenticated. The client application will be authenticated, so it will be able to access the Web API.

authorized to access the Web API

---
Refers to: 4-WebApp-your-API/4-2-B2C/README.md:141 in ba50129. [](commit_id = ba50129f4afd0de7ee83e152e019fdc117f5fc8e, deletion_comment = False)

**Comment by kalyankrishna1** [comment]
> Explore the sample by signing in into the TodoList client, adding items to the To Do list. If you stop the application without signing out, the next time you run the application, you won't be prompted to sign in again.

ToDo

---
Refers to: 4-WebApp-your-API/4-2-B2C/README.md:143 in ba50129. [](commit_id = ba50129f4afd0de7ee83e152e019fdc117f5fc8e, deletion_comment = False)

**Comment by kalyankrishna1** [comment]
>     dotnet new mvc --auth SingleOrg --client-id <Enter_the_Application_Id_here> --tenant-id <yourTenantId>

Does this work for B2C tenants too?

---
Refers to: 4-WebApp-your-API/4-2-B2C/README.md:160 in ba50129. [](commit_id = ba50129f4afd0de7ee83e152e019fdc117f5fc8e, deletion_comment = False)

File: 4-WebApp-your-API/4-2-B2C/README.md [hunk 6]
```diff
@@ -384,23 +386,23 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
 
 For more information, visit the following links:
```
Comment (kalyankrishna1) [inline 4-WebApp-your-API/4-2-B2C/README.md:387]: No links to B2C docs??

Comment (Shama-K) [reply 4-WebApp-your-API/4-2-B2C/README.md]: Added links.
Thanks.

**Comment by Shama-K** [comment]
> > ```
> > dotnet new mvc --auth SingleOrg --client-id <Enter_the_Application_Id_here> --tenant-id <yourTenantId>
> > ```
> 
> Does this work for B2C tenants too? Note that the end user would NOT have a B2C tenant. So even if it can be later modified, do check if this command will work with a B2C tenantID
> 
> Refers to: 4-WebApp-your-API/4-2-B2C/README.md:160 in [ba50129](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/commit/ba50129f4afd0de7ee83e152e019fdc117f5fc8e). [](commit_id = [ba50129](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/commit/ba50129f4afd0de7ee83e152e019fdc117f5fc8e), deletion_comment = False)

The command works and creates sample. As it is SingleOrg template, it creates **AzureAd** section in appsettings.json.

**Review by kalyankrishna1 — APPROVED** [review]

**Pull request merged**